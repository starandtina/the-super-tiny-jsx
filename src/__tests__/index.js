const { compiler, tag } = require('../index')

it('compiler', () => {
  expect(
    compiler(`
      html {
        /* multi line comment */
        body {
          h1 class="h1" style="color: red" {
            123456789 // single line comment
            "Hello World!" 
          }
          button disabled type="submit" {
            "Submit"
          }
        }
      }
      `),
  ).toMatchSnapshot()
})

it('tag', () => {
  const ttttest = 'ttttest'
  const arr = ['xx', 'oo']

  expect(
    tag`
      html {
        /* comments */
        body {
          h1 {
            "Hello World! - ${ttttest}"
          }
          p {
            "${arr}"
          }
        }
      }
    `,
  ).toMatchSnapshot()
})
