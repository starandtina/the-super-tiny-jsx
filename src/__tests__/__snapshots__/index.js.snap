// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler 1`] = `
"<!DOCTYPE html>
<html>
  <!-- multi line comment  -->
  <body>
    <h1 class=\\"h1\\" style=\\"color: red\\">
      123456789
      <!-- single line comment  -->
      Hello World!
    </h1>
    <button disabled=\\"true\\" type=\\"submit\\">
      Submit
    </button>
  </body>
</html>"
`;

exports[`tag 1`] = `
"<!DOCTYPE html>
<html>
  <!-- comments  -->
  <body>
    <h1>
      Hello World! - ttttest
    </h1>
    <p>
      xxoo
    </p>
  </body>
</html>"
`;
